### DISCUSSION TITLE
Request a feature of "Open Containing Folder in Explorer"

### DISCUSSION THREAD

**Discussion by oceanfish567** [apache/netbeans#7074] [Ideas]
Open Containing Folder in Explorer. It provides a very convenient and fast shortcut to achieve the goal.

### 1. example from notepad++
![p102](https://github.com/apache/netbeans/assets/103805617/39ce56bd-a731-4978-a978-66d64188210f)

### 2. This is the result.
![p103](https://github.com/apache/netbeans/assets/103805617/f6693267-1feb-40a8-9a4c-4a3add313b66)

### 3. I want this feature to be available in netbeans.
![p104](https://github.com/apache/netbeans/assets/103805617/30db1aef-f7bc-4bf4-a554-e23c9d58e098)

![p105](https://github.com/apache/netbeans/assets/103805617/519b2bf8-a5e8-48c4-a756-a903d6c80305)

**Comment by junichi11** [comment]
Maybe, this: https://plugins.netbeans.apache.org/catalogue/?id=76

**Reply by oceanfish567** [reply]
Thank you very much.

**Comment by neilcsmith-net** [comment]
While it's not quite the same, you can open the parent folder in the system file browser from the parent popup in the project, files or favourites views. Or by `ALT-SHIFT-S`.

I like the look of the option in notepad++ to open in a system terminal too. I keep meaning to look at something like that.

**Reply by mbien** [reply]
https://github.com/apache/netbeans/pull/7081 I put it next to the other action since it is in the tools sub menu already in some cases.

**Reply by Chris2011** [reply]
The problem here is, first we have "open in system" which you can find in tools (I dunno why it is located there by anyway) and problem two is, that it will not open in explorer, it will open in the related application. I find it strange, because the action is for folders which opens the folders. We should think about it. I already have this plugin owend and I there is two actions for it: https://github.com/Chris2011/QuickOpener-NetBeans (open in explorer (folder or file doesn't matter), open external terminal, open in related application or custom command

**Reply by neilcsmith-net** [reply]
@Chris2011 What's strange about it? It opens the selected file or folder in the system application that handles it.

**Reply by Chris2011** [reply]
I had an other impression. I know all of those "open in explorer" and as far as I know there was a PR to just implement this for a folder, which is working. But afaik it was about opening the folder. accidentally I saw this action for files too and my impression was "open in system" is opening the folder of the file.

**Reply by neilcsmith-net** [reply]
The existing action working on files has been there a long time (pre-ASF), but was previously only registered for unknown file types.  This was changed to be registered for all files and folders in #5770 , with some additional fixes and shortcut (`ALT+SHIFT+S`) added in #5870 and #6159.

Having another action that opens the containing folder would be good, but this is not what the existing action was for.

**Reply by Chris2011** [reply]
Thx for clarification.

**Reply by Chris2011** [reply]
It just came into my view now, that you already mentioned this ticket: #5770 and the title is exact what I was thinking of: "Added the option to open the selected file or directory in the system file explorer". Maybe I missunderstood the title, for me it was like both (file or directory) in system file explorer. And also this commenter wasn't sure what happens: https://github.com/apache/netbeans/pull/5770#issuecomment-1773095719 it is just about that it is not 100% clear.

Anyway I will try to work on an improvement to make it more clear and also to add both (open in app/open containing folder).

**Reply by mbien** [reply]
https://github.com/apache/netbeans/pull/7081
